DataManagementSystem/DB/FileCatalogComponents/DirectoryMetadata.py

```diff
@@ -24,25 +25,52 @@ def setDatabase(self, database):
 #  Manage Metadata fields
 #
 
+  def getMetaName(self, meta, credDict):
+    """
+    Return a metadata name based on client supplied meta name and client credentials
+    For the base class it just returns the name passed in.
+    This method is meant to be overwritten by derived classes.
+
+    :param meta:  meta name
+    :param credDict: client credentials
+    :return: meta name
+    """
+
+    return meta
```

Comment:
What about returning `S_ERROR("method to be overwritten")` ?

Reply:
This method is a hook, and it works "as is" w/o overwriting it. I added an underscore to the method name to make it clear that it is an internal method.

Reply:
Why do you need at all getMetaName and getMetaNameSuffix in the FC classes ? Just use methods from the Utilities module. The same for the File metadata, the utility functions are the same.

Reply:
The functions can get argument specifying the naming method (no modifiction, by group, by VO, by ...). This can be defined in the catalog configuration. Then you do not need inherited classes to just implement this function